src/bytes.rs

```diff
@@ -63,6 +79,16 @@ impl ByteBuf {
             bytes: Vec::with_capacity(cap)
         }
     }
+
+    pub fn as_vec(self) -> Vec<u8> {
+        self.bytes
+    }
+}
```

Comment:
does removing the `From` impl and adding the `Into` impl work?
Then the `From` impl is generated in the standard library if I read this correctly.